### PR TITLE
fixed 'count' to 'entries.count' in `aws_ec2_security_groups` as per the docs

### LINKED ISSUE #141: `count` is not implemented in `aws_ec2_security_groups
easy fix

### LINKED ISSUE #148: Integration tests fail due to floating terraform aws plugin version
Running the integration tests on a freshly cloned repo with terraform 0.10.0 gives this result:

```
Initializing provider plugins...
- Downloading plugin for provider "aws"...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.aws: version = "~> 1.6"
```
*snip*

```
You're now on a new, empty workspace. Workspaces isolate their state,
so if you run "terraform plan" Terraform will not see any existing state
for this configuration.
cd /Users/sam/projects/devsecops/inspec-aws/test/integration/default/build/ && AWS_PROFILE=inspec-aws-test-default terraform plan
Error asking for user input: 1 error(s) occurred:

* provider.aws: dial unix /var/folders/vj/_krnh7zj7l3d46z485bml_j00000gp/T/plugin261050608|netrpc: connect: no such file or directory
Integration testing has failed for the default account
----> Cleanup
cd /Users/sam/projects/devsecops/inspec-aws/test/integration/default/build/ && AWS_PROFILE=inspec-aws-test-default terraform destroy -force
Error asking for user input: 1 error(s) occurred:

* provider.aws: dial unix /var/folders/vj/_krnh7zj7l3d46z485bml_j00000gp/T/plugin005393920|netrpc: connect: no such file or directory
```

I had to step down the version of the aws provder in the aws.tf file until the tests were able to run successfully to completion.

```
provider "aws" {
  version = "= 1.1"
}
```

I was previously able to get things basically working by simply changing to the aws.tf file to use terraform 0.11.1, and was then able to use newest version of the aws provider (1.6).  Note that if you had previously installed aws provider 1.1, and you are using terraform 0.10.0, you don't need to hack the aws.tf file because terraform won't upgrade the provider unless it thinks it needs to.

### PR DESCRIPTION
**Pull request by aaronlippold**
Fixes #141 
Fixes #148

### CODE REVIEW

File: test/integration/default/build/aws.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
 terraform {
-  required_version = "~> 0.10.0"
```
Comment (clintoncwolfe) [inline test/integration/default/build/aws.tf]: This seems to be an unrelated change.  Terraform changes a lot on minor version bumps; our pinning to v0.10.x is intentional, to avoid breakage when testers use incompatible versions.  If you have a different verision of terraform, you might consider using tfenv, which allows you to switch versions easily.

If you feel this is a needed change, please open a separate issue/PR.

File: test/integration/default/verify/controls/aws_ec2_security_groups.rb [hunk 2]
```diff
@@ -40,4 +40,11 @@
...
     its('group_ids') { should include fixtures['ec2_security_group_default_group_id'] }
   end
-end
\ No newline at end of file
+end
+
+control "aws_ec2_security_groups" do
+  # Verify you have more than the default security group
+  describe aws_ec2_security_groups do
+    its('count') { should be >= 2 }
```
Comment (clintoncwolfe) [inline test/integration/default/verify/controls/aws_ec2_security_groups.rb]: The documented property is `'entries.count'`.  Adding a test for it is a great idea, though!

File: test/integration/minimal/build/aws.tf [hunk 3]
```diff
@@ -1,10 +1,10 @@
 terraform {
-  required_version = "~> 0.10.0"
+  required_version = "> 0.10.0"
```
Comment (clintoncwolfe) [inline test/integration/minimal/build/aws.tf]: See above about TF version pinning.

File: libraries/aws_ec2_security_groups.rb [hunk 4]
```diff
@@ -25,6 +25,10 @@ def access_key_data
     @table
   end
 
+  def count
```
Comment (clintoncwolfe) [inline libraries/aws_ec2_security_groups.rb]: I do really like the idea of exposing a `count` property directly rather than via `entries`.  This could be pushed up to a mixin for plural resources one day.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]

File: libraries/aws_ec2_security_groups.rb (hunk 4 above)
Comment (aaronlippold) [reply libraries/aws_ec2_security_groups.rb]: Resolved in last push

File: test/integration/default/build/aws.tf (hunk 1 above)
Comment (aaronlippold) [reply test/integration/default/build/aws.tf]: Fixed and added pinning to the AWS provider so the two linking play well together.

File: test/integration/minimal/build/aws.tf (hunk 3 above)
Comment (aaronlippold) [reply test/integration/minimal/build/aws.tf]: fixed

File: test/integration/default/verify/controls/aws_ec2_security_groups.rb (hunk 2 above)
Comment (aaronlippold) [reply test/integration/default/verify/controls/aws_ec2_security_groups.rb]: updated this to 'entries.count' as per the docs.

**Review by aaronlippold — COMMENTED** [review]
I addressed all of @clintoncwolfe comments and things should be good to go now.

**Review by clintoncwolfe — APPROVED** [review]

**Comment by clintoncwolfe** [comment]
Thanks for the fix for #141 - for updating the example in the docs, and adding an integration test.
